packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -178,6 +181,127 @@ class SegmentedButton<T> extends StatefulWidget {
+  /// [SegmentedButton], as well as its overlay color, with all of the
+  /// standard opacity adjustments for the pressed, focused, and
+  /// hovered states, one could write:
+  ///
+  /// ```dart
+  /// SegmentedButton<int>(
+  ///   style: SegmentedButton.styleFrom(foregroundColor: Colors.green),
+  ///   segments: const <ButtonSegment<int>>[
+  ///      ButtonSegment<int>(
+  ///          value: 0, label: Text('0'), icon: Icon(Icons.calendar_view_day)),
+  ///      ButtonSegment<int>(
+  ///          value: 1, label: Text('1'), icon: Icon(Icons.calendar_view_week)),
+  ///      ButtonSegment<int>(
+  ///          value: 2, label: Text('2'), icon: Icon(Icons.calendar_view_month)),
+  ///      ButtonSegment<int>(
+  ///          value: 3, label: Text('3'), icon: Icon(Icons.calendar_today)),
+  ///    ],
+  ///    selected: const <int>{0},
+  ///    onSelectionChanged: (Set<int> selection) {},
+  /// ),
+  /// ```
+  static ButtonStyle styleFrom({
+    Color? foregroundColor,
+    Color? selectedForegroundColor,
+    Color? selectedBackgroundColor,
+    Color? backgroundColor,
+    Color? disabledForegroundColor,
+    Color? disabledBackgroundColor,
+    Color? shadowColor,
+    Color? surfaceTintColor,
+    double? elevation,
+    TextStyle? textStyle,
+    EdgeInsetsGeometry? padding,
+    Size? minimumSize,
+    Size? fixedSize,
+    Size? maximumSize,
+    BorderSide? side,
+    OutlinedBorder? shape,
+    MouseCursor? enabledMouseCursor,
+    MouseCursor? disabledMouseCursor,
+    VisualDensity? visualDensity,
+    MaterialTapTargetSize? tapTargetSize,
+    Duration? animationDuration,
+    bool? enableFeedback,
+    AlignmentGeometry? alignment,
+    InteractiveInkFeatureFactory? splashFactory,
+    Color? primary,
+    Color? onPrimary,
+    Color? onSurface,
+  }) {
+    final Color? background = backgroundColor ?? primary;
+    selectedBackgroundColor ??= background;
+    final Color? disabledBackground = disabledBackgroundColor ?? onSurface?.withOpacity(0.12);
+    final MaterialStateProperty<Color?>? backgroundColorProp = (background == null && disabledBackground == null)
+      ? null
+      : _SegmentButtonDefaultColor(background, disabledBackground, selectedBackgroundColor);
+    final Color? foreground = foregroundColor ?? onPrimary;
+    selectedForegroundColor ??= foreground;
+    final Color? disabledForeground = disabledForegroundColor ?? onSurface?.withOpacity(0.38);
+    final MaterialStateProperty<Color?>? foregroundColorProp = (foreground == null && disabledForeground == null)
+      ? null :
+      _SegmentButtonDefaultColor(foreground, disabledForeground, selectedForegroundColor);
+    final MaterialStateProperty<Color?>? overlayColor = (foreground == null)
+      ? null
+      : _SegmentedButtonDefaultOverlay(foreground);
+    final MaterialStateProperty<double>? elevationValue = (elevation == null)
+      ? null
+      : _SegmentedButtonDefaultElevation(elevation);
```

Comment:
Default elevation is `0` for all the material states so we can just return provided `elevation` for all states.

```suggestion
    final MaterialStateProperty<double>? elevationValue = (elevation == null)
      ? null
      : MaterialStatePropertyAll<double>(elevation);
```